### ISSUE TITLE
All JSON filters fail with `fd:4: hPutBuf: resource vanished (Broken pipe)`

### ISSUE THREAD

**Issue by stroobandt** [jgm/pandoc#3296]
With
```
$ pandoc --version
pandoc 1.19
Compiled with pandoc-types 1.17.0.4, texmath 0.9, highlighting-kate 0.6.3
```
as well as `pandoc 1.16.0.2` on `Ubuntu 16.04.1 LTS`, I receive the following error:
```
pandoc: Error running filter ./behead2.hs
fd:4: hPutBuf: resource vanished (Broken pipe)
```
when running [the example `behead2.hs`](http://pandoc.org/scripting.html#json-filters) JSON filter
```
$ pandoc -o file.html --filter ./behead2.hs file.md
```
with
```
#!/usr/bin/env runhaskell
-- behead2.hs
import Text.Pandoc.JSON

main :: IO ()
main = toJSONFilter behead
  where behead (Header n _ xs) | n >= 2 = Para [Emph xs]
        behead x = x
```
or any other JSON filter that used to work with `pandoc 1.13` under (X)Ubuntu 14.04 LTS.
"_I feel beheaded…_"

**Comment by jgm** [comment]
The JSON serialization format changed (as you can see in
the changelog for 1.18).

What this means is that, if you use a filter with pandoc
1.18+, you need to make sure it is linked against
pandoc-types 1.17+.

You may have an older version installed.  Try `ghc-pkg list
pandoc-types`.  You may need to `cabal install
pandoc-types`.

**Comment by stroobandt** [comment]
@jgm Thank you, John, for the excellent explanation.

For those reading this, I solved my issue on (X)Ubuntu LTS 16.04 running `pandoc 1.16.0.2` by installing the following additional package:

    $ sudo apt install libghc-pandoc-prof

For newer `pandoc` versions, the name of this package can be different, as indicated above.
The takeaway message is: _**Do not mix package versions!**_
